File: static/libs/facebook.js

var fb_id;

$(document).ready(() => {
    setInterval(function(){ reload_page(); },5000);
    $.get("/api/user", (user, err) => {
        if (err !== "success") console.error(err);
        if (user && Array.isArray(user.data)) {
            var username_array = [...new Set(user.data.map(user => user.fb_name))]
            var fbid_array = [...new Set(user.data.map(user => user.fb_id))]
            const login_container = document.querySelector(".loginname")
            var login_name = username_array[0]
            fb_id = fbid_array[0]
            login_container.innerText = "WELCOME " + login_name.toUpperCase()
        }
    $.get("/api/facebook/" + fb_id, (facebook, err) =>{
        if (err !== "success") console.error(err);
        if (facebook && Array.isArray(facebook.data)) {
            var post = '';
            facebook_data = sortByKeyDesc(facebook.data, 'created_time')
            for (var i=0; i<facebook_data.length; i++)
            {
                var url_li = '';
                if (facebook_data[i]['attach_url'] != 'None')
                {
                   url_li = '<li> Url: <a href="' + facebook_data[i]['attach_url']+ '">Attachment</a></li>'
                }
                if(facebook_data[i]['attach_url'] == 'None')
                {
                    url_li = '<li> Url: None</li>'
                }
                post = post + '<div class="innerpost"><ul>'
                            +'<li> Create Date: ' + (facebook_data[i]['created_time'].substring(0,facebook_data[i]['created_time'].indexOf('+'))).replace('T',' ') + '</li>'
                            +'<li> Status Type: ' + (facebook_data[i]['status_type'].split('_').join(' ')).replace(/\b\w/g, function(l){ return l.toUpperCase() }) + '</li>'
                            +'<li> Caption: ' + facebook_data[i]['caption'] + '</li>'
                            +'<li> Message: ' + facebook_data[i]['message'] + '</li>'
                            +'<li> Story: ' + facebook_data[i]['story'] + '</li>'
                            +'<li> Description: ' + facebook_data[i]['description'] + '</li>'
                            +'<li> Type: ' + facebook_data[i]['type'].replace(/\b\w/g, function(l){ return l.toUpperCase() }) + '</li>'
                            +'<li> Attachement: <div><ul>' 
                                +'<li> Title: ' + facebook_data[i]['attach_title']+ '</li>'
                                +'<li> Type: ' + (facebook_data[i]['attach_type'].split('_').join(' ')).replace(/\b\w/g, function(l){ return l.toUpperCase() })+ '</li>'
                                + url_li
                                +'<li> Description: ' + facebook_data[i]['attach_description']+ '</li>'
                                +'</ul></div></li>'
                            +'</ul></div>'
            }
            $(".posts").append(post)
        }
    })
})
});

function sortByKeyDesc(array, key) {
    return array.sort(function (a, b) {
        var x = a[key]; var y = b[key];
        return ((x > y) ? -1 : ((x < y) ? 1 : 0));
    });
}

function sortByKeyAsc(array, key) {
    return array.sort(function (a, b) {
        var x = a[key]; var y = b[key];
        return ((x < y) ? -1 : ((x > y) ? 1 : 0));
    });
}

function reload_page()
 {
    window.location.reload(true);
 }

File: models/facebook.py
from sqlalchemy import Column, Integer, String, Date
from models.base import Base
import datetime

#### User class to create User table
class Facebook(Base):
    __tablename__ = 'facebook'

    id = Column(String, primary_key=True)
    fb_id = Column(String)
    status_type = Column(String)
    caption = Column(String)
    story = Column(String)
    message = Column(String)
    description = Column(String)
    type = Column(String)
    attach_url = Column(String)
    attach_title = Column(String)
    attach_type = Column(String)
    attach_description = Column(String)
    created_time = Column(String)
    inserted_date = Column(Date(), default=datetime.datetime.now().date())

    def __repr__(self):
        return "<Facebook (id='%s', fb_id='%s', status_type='%s', caption='%s', story='%s', message='%s', description='%s', type='%s', attach_url='%s', attach_title='%s', attach_type='%s', attach_description='%s', created_time='%s', inserted_date='%s' )" % (self.id, self.fb_id, self.status_type, self.caption, self.story, self.message, self.description, self.type, self.attach_url, self.attach_title, self.attach_type, self.attach_description, self.created_time, self.inserted_date)
